### PR TITLE
Fix doc for rustc "-g" flag

### PR DESCRIPTION
**Pull request by TheBiggerGuy**
The rustc `-g` CLI flag was miss documented to be a synonym of `-C debug-level=2` and not `-C debuglevel=2`. Also add links to the codegen docs for each synonym.

I am unsure of this will conflict with work on #52938

### CODE REVIEW

File: src/doc/rustc/src/command-line-arguments.md [hunk 1]
```diff
@@ -42,11 +42,11 @@ This flag prints out various information about the compiler.
 
 ## `-g`: include debug information
 
-A synonym for `-C debug-level=2`.
+A synonym for `-C debuglevel=2`, for more see [here](codegen-options/index.html#debuginfo).
```
Comment (TimNN) [inline src/doc/rustc/src/command-line-arguments.md]: Looking at the output of `rustc +stable -C help`, it looks like this should actually be `debuginfo`.

**Review by TimNN — CHANGES_REQUESTED** [review]
One nit, otherwise LGTM.

**Comment by GuillaumeGomez** [comment]
I don't get why I was cced in here since there are no changes in HTML/CSS/JS but whatever... Looks good to me, thanks!

@bors: r=TimNN

**Comment by TheBiggerGuy** [comment]
Sorry for the first version not having the correct arg. A bit of a fail trying to fix a typo with another one.

I have just ran the full set of tests and check the doc links work locally.

The only question I have is if ```debug-level``` or ```debuglevel``` where ever correct? If so maybe a sentence in the doc letting people know of the rename.

**Comment by TheBiggerGuy** [comment]
My quick check of the code found that LLVM uses `debug-level` and rustc uses `debuginfo`:

rustc
------
https://github.com/rust-lang/rust/blob/f47ec2ad5b6887b3d400aee49e2294bd27733d18/src/librustc/session/config.rs#L1684-L1685

llvm
-----
https://github.com/rust-lang/llvm-project/blame/73a75d35b9776d56160fa3200aca4a970ae49b60/llvm/docs/CommandLine.rst#L420-L421

**Pull request merged**